IRCConnector.py

```diff
@@ -30,17 +37,20 @@ def run(self):
 
 
 class IRCLego(Lego):
-    def __init__(self, channel, nickname, server, baseplate, port=6667):
-        super(Lego, self).__init__(baseplate)
-        self.botThread = IRCBot(channel, nickname, server, baseplate, port)
+    def __init__(self, channel, nickname, server, baseplate, lock, port=6667, use_ssl=False):
```

Comment:
Channel, nickname, and server are specific to the user's use case. Baseplate and Lock must be passed in from the Legobot file because they are shared between multiple Legos.